Bio/Seq.py

```diff
@@ -2452,6 +2558,21 @@ def toseq(self):
         """
         return Seq("".join(self.data), self.alphabet)
 
+    def join(self, other):
+        """Return a merge of the sequences in other, spaced by the sequence from self.
+
+        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+
+        >>> concatenated = MutableSeq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable
+        are not Seq or String objects
+        """
+        seq_joined = self.toseq().join(other)  # returns Seq object instead of MutableSeq
+        return seq_joined
```

Comment:
It's not clear to me, why you return a ```Seq``` object here, instead of a ```MutableSeq```. Somewhere in the discussion you said that you tried to be consistent with the ```add``` method (you mean ```__add__```?), but for ```MutableSeq``` ```__add__``` returns a new ```MutableSeq``` object.

Reply:
An earlier iteration did return a ``MutableSeq``, but required a lot of code duplication - returning a ``Seq`` avoided that neatly.

Reply:
But can't you finally make a new ```MutableSeq``` with the joined sequence and alphabet and return this? E.g. using ```tomutable()```. Like in the ```__add__``` method?

On the other hand, I wonder if I would expect the result of a join operation, where the MutableSeq acts "only" as spacer, to be a MutableSeq. So this is probably fine, but should be mentioned in the docstring.

Reply:
You could argue either way on the the return type, personally I think a ``Seq`` is more likely to be useful as when I have needed a ``MutableSeq`` it has only been briefly. The doctest does show a ``Seq``, but you are right, saying so in the text of the docstring would be clearer.